lte/gateway/c/session_manager/StatsPoller.cpp

```diff
@@ -13,17 +13,32 @@
 #include <stdint.h>
 #include <atomic>
 #include <memory>
+#include <vector>
 #include "StatsPoller.h"
-#define COOKIE 0
-#define COOKIE_MASK 0
+#define OVS_COOKIE_MATCH_ALL 0xffffffff
 
 namespace magma {
 
 void StatsPoller::start_loop(
     std::shared_ptr<magma::LocalEnforcer> local_enforcer,
     uint32_t loop_interval_seconds) {
+  // when no shard ids are available poll all UEs
+  // to poll by cookie pass in a vector of shard ids, where
+  // each element is a shard id that maps directly to a cookie
+  auto shard_tracker = local_enforcer->get_shard_tracker();
   while (true) {
-    local_enforcer->poll_stats_enforcer(COOKIE, COOKIE_MASK);
+    // TODO(veshkemburu): add smart polling function to determine which
+    // shard ids to poll
+    if (shard_tracker.get_size_of_tracker() == 0) {
+      local_enforcer->poll_stats_enforcer(0, 0);
+    } else {
+      // iterate through all shard ids, but only poll shard IDs that have UEs
+      for (size_t i = 0; i < shard_tracker.get_size_of_tracker(); i++) {
+        if (!shard_tracker.get_imsis_per_shard().at(i).empty()) {
```

Comment:
This feels, complicated. Lets just add a function to shard_tracker that returns a vector of active shards?

Reply:
I was thinking about doing that, and frankly I like that idea much better.